substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -734,11 +897,13 @@ impl<T: Config> Pallet<T> {
         Ok(contract)
     }
 
-    pub fn _update_node_contract(
+    #[transactional]
+    pub fn _update_deployment_contract(
         account_id: T::AccountId,
         contract_id: u64,
         deployment_hash: DeploymentHash,
         deployment_data: DeploymentDataInput<T>,
+        extra_resources: Option<Resources>,
```

Comment:
While the intention was good, this must be generic and just pass resources. In the body of the function, check if the resources requested are higher than what is currently saved on the contract. Else, the client will have to calculate this and this might be weird.